### ISSUE TITLE
Forward process errors in `ess-command` as Emacs errors

### ISSUE THREAD

**Issue by lionel-** [emacs-ess/ESS#1113]
Currently the R errors are silently ignored.

**Comment by vspinu** [comment]
Maybe just messages. Emacs errors is too much, especially if debug-on-error is on.

**Comment by lionel-** [comment]
With `quit-flag` set then? `ess-command` should abort the current evaluation, there is no point continuing evaluation and returning to the caller when an error happened.

**Comment by lionel-** [comment]
I think errors are the correct design. If an error is likely to happen on the R side and should be ignored then that should be taken care of by condition handlers on the R side. That is already how many ESSR commands which swallow errors with `tryCatch()` are designed. It's also possible to handle the errors on the Emacs side with `ignore-errors` if needed.

**Comment by vspinu** [comment]
It probably should be considered case by case. Cascading is a concern. When ESSR load errors and a fallback of sourcing is attempted we don't want to throw in the middle, but would still want to inform the user. This is why MPI framework has an option to convey errors, warnings and messages to emacs.

**Comment by lionel-** [comment]
Right, we'd bubble up R warnings and messages as Emacs messages on the Emacs side. It would be up to the ESSR loading procedure to avoid throwing on the R side. Or if it does, we could catch the error on the Emacs side and attempt sourcing. Bubbling up conditions from `ess-command` provides more ways of structuring code via exception handling.

We could also transform R interrupts to Emacs quits. Other kinds of early exits (long returns, restart invokations) should be rare and should probably be turned to errors? Hmm on recent R versions there could also be global calling handlers catching conditions. If that proves to be a problem, we can disable those temporarily.
